### PR TITLE
Fedora gues caps

### PR DESCRIPTION
**Pull request by humaton**

### CODE REVIEW

File: plugins/guests/fedora/cap/flavor.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+module VagrantPlugins
+  module GuestFedora
+    module Cap
+      class Flavor
+        def self.flavor(machine)
+          # Read the version file
+          output = ""
```
Comment (strzibny) [inline plugins/guests/fedora/cap/flavor.rb]: `output` is redundant

File: plugins/guests/fedora/cap/flavor.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          machine.communicate.sudo("grep VERSION_ID /etc/os-release") do |type, data|
+            version = data.split("=")[1].chomp.to_i if type == :stdout
+          end
+
+          # Detect various flavors we care about
+          if version >= 20
+            return :"fedora_#{version}"
+          else if version <= 16
```
Comment (strzibny) [inline plugins/guests/fedora/cap/flavor.rb]: It was already pointed out by voxik that we don't need two same returns.

File: lib/vagrant-registration.rb [hunk 3]
```diff
@@ -15,6 +15,8 @@ def self.source_root
       @source_root ||= Pathname.new(File.expand_path("../../", __FILE__))
     end
 
+    # Temporally load the extra capability files for Fedora
+    load(File.join(self.source_root, 'plugins/guests/fedora/plugin.rb'))
```
Comment (strzibny) [inline lib/vagrant-registration.rb]: We should still consider loading these files according to the current machine's OS (guest).

Comment (humaton) [inline lib/vagrant-registration.rb]: I think that is kind of anti pattern, this is temp. solution until we get guest changes to default vagrant plugins.

Comment (strzibny) [inline lib/vagrant-registration.rb]: Well exactly because of that. It also still lacks checks on whether those files are shipped already or not...

**Pull request merged**